src/workerd/io/hibernation-manager.c++

```diff
@@ -178,6 +173,9 @@ kj::Promise<void> HibernationManagerImpl::readLoop(HibernatableWebSocket& hib) {
       }
       KJ_CASE_ONEOF(close, kj::WebSocket::Close) {
         maybeParams.emplace(close.code, kj::mv(close.reason), true);
+        // We'll dispatch the close event, so let's mark our websocket as having done so to
+        // prevent a situation where we dispatch it twice.
+        hib.hasDispatchedClose = true;
```

Comment:
It was like this before, I changed it to be after the event returned when we switched to coroutines. It's mostly paranoia that somehow the issue was we dispatched a close via normal readLoop, released the websocket but didn't complete the promise, then tried to dispatch it again in the disconnect handler (which would assert we are `Accepted`).

Reply:
Something like this seems more likely to cause trouble than the `watchAbortTask()` running since I imagine it would run the same when websockets were active